client/state/posts/utils.js

```diff
@@ -21,22 +21,31 @@ export function getNormalizedPostsQuery( query ) {
 
 /**
  * Returns a serialized posts query, used as the key in the
- * `state.posts.siteQueries` state object.
+ * `state.posts.queries` state object.
  *
- * @param  {Object} query Posts query
- * @return {String}       Serialized posts query
+ * @param  {Object} query  Posts query
+ * @param  {Number} siteId Optional site ID
+ * @return {String}        Serialized posts query
  */
-export function getSerializedPostsQuery( query = {} ) {
-	return JSON.stringify( getNormalizedPostsQuery( query ) ).toLowerCase();
+export function getSerializedPostsQuery( query = {}, siteId ) {
+	const normalizedQuery = getNormalizedPostsQuery( query );
+	const serializedQuery = JSON.stringify( normalizedQuery ).toLocaleLowerCase();
+
+	if ( siteId ) {
+		return [ siteId, serializedQuery ].join( ':' );
+	}
+
+	return serializedQuery;
 }
 
 /**
  * Returns a serialized posts query, excluding any page parameter, used as the
- * key in the `state.posts.siteQueriesLastPage` state object.
+ * key in the `state.posts.queriesLastPage` state object.
  *
- * @param  {Object} query Posts query
- * @return {String}       Serialized posts query
+ * @param  {Object} query  Posts query
+ * @param  {Number} siteId Optional site ID
+ * @return {String}        Serialized posts query
  */
-export function getSerializedPostsQueryWithoutPage( query ) {
-	return JSON.stringify( omit( getNormalizedPostsQuery( query ), 'page' ) ).toLowerCase();
+export function getSerializedPostsQueryWithoutPage( query, siteId ) {
+	return getSerializedPostsQuery( omit( query, 'page' ), siteId );
```

Comment:
>Don't forget to update the lodash require above.

This file should have been captured by the update, and not really a problem since I've not revised the dependencies, but I'll do a rebase to be safe.